modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/portlet/MonitoringInvokerPortlet.java

```diff
@@ -348,10 +347,56 @@ else if (e instanceof PortletException) {
 		}
 	}
 
+	private void _render(
+			RenderRequest renderRequest, MimeResponse mimeResponse,
+			Renderable renderable)
+		throws IOException, PortletException {
+
+		DataSample dataSample = null;
+
+		try {
+			if (_portletMonitoringControl.isMonitorPortletRenderRequest()) {
+				dataSample = _dataSampleFactory.createPortletRequestDataSample(
+					PortletRequestType.RENDER, renderRequest, mimeResponse);
```

Comment:
@ngriffin7a so we are not creating a new PortletRequestType.HEADER to tell the difference between render and header monitoring?

Reply:
@shuyangzhou Good catch -- `PortletRequestType.HEADER` was an oversight. Please add it in your code review or let me know if you would like me to do it.